src/OrbitGl/CaptureWindow.cpp

```diff
@@ -814,3 +821,41 @@ void CaptureWindow::RenderSelectionOverlay() {
   text_renderer_.AddText(text.c_str(), stop_pos_world, select_stop_pos_world_[1],
                          GlCanvas::kZValueOverlay, formatting);
 }
+
+void CaptureWindow::DrawLayerDebugInfo(
+    const std::vector<orbit_gl::BatchRenderGroupId>& sorted_groups, QPainter* painter) {
+  QFont font = QFontDatabase::systemFont(QFontDatabase::GeneralFont);
+
+  for (const auto& group : sorted_groups) {
+    const orbit_gl::StencilConfig& stencil =
+        render_group_manager_.GetGroupState(group.name).stencil;
+    if (!stencil.enabled) continue;
+
+    PrepareGlState();
+    if (time_graph_ != nullptr) {
+      const Color color = time_graph_->GetColor(static_cast<uint32_t>(
+          std::hash<std::string>()(group.name) % std::numeric_limits<uint32_t>::max()));
+      glColor4f(static_cast<float>(color[0]) / 255.f, static_cast<float>(color[1]) / 255.f,
+                static_cast<float>(color[2]) / 255.f, 1.f);
+    } else {
+      glColor4f(1, 0, 0, 1);
+    }
+    glBegin(GL_POLYGON);
```

Comment:
could we use the batcher here instead of glbegin/glend?

Reply:
Hm I think that would make things more complicated than needed - the Batcher is still used in this mode to collect all primitives, but I'm omitting the render calls. I'd rather keep this debug functionality out of the batcher and keep it like this, the CaptureWindow / GlCanvas already uses raw OpenGl (e.g. to prepare the viewport and state), so I'd argue it's not outside of their scope. WDYT?